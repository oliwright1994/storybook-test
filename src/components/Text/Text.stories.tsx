import { Meta } from '@storybook/react/types-6-0'
import React from 'react'

import Stack from '../Stack'
import Text, { IText } from './Text'

export default { component: Text, title: 'Text' } as Meta
export const Standard = (args: IText): JSX.Element => (
  <Stack>
    <Text {...args}>Text</Text>
  </Stack>
)
Standard.args = {
  as: 'p',
}
