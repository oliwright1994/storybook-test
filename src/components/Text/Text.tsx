import { Text as ChakraText } from '@chakra-ui/core'
import React, { FC } from 'react'

export interface IText {
  as?: 'p' | 'span' | 'strong' | 'i'
}

const Text: FC<IText> = ({ children, as = 'p' }) => {
  return <ChakraText as={as}>{children}</ChakraText>
}

export default Text
