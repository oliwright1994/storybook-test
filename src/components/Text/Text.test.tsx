import { screen } from '@testing-library/react'
import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Text from '.'

describe('<Text />', () => {
  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(
      <>
        <Text>Test One</Text>
        <Text>Test Two</Text>
        <Text>Test Three</Text>
      </>
    )
    const textNodes = screen.getAllByText(/test/i)

    expect(textNodes).toHaveLength(3)
    expect(container).toMatchSnapshot()
  })

  it('should render with the correct element tag', () => {
    const { container } = renderWithBrightTheme(
      <>
        <Text>Paragraph</Text>
        <Text as="i">Italic</Text>
        <Text as="strong">Strong</Text>
        <Text as="span">Span</Text>
      </>
    )
    expect(screen.getByText(/paragraph/i).nodeName).toEqual('P')
    expect(screen.getByText(/i/i).nodeName).toEqual('I')
    expect(screen.getByText(/strong/i).nodeName).toEqual('STRONG')
    expect(screen.getByText(/span/i).nodeName).toEqual('SPAN')
    expect(container).toMatchSnapshot()
  })
})
