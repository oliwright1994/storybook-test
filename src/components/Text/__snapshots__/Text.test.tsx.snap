// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render as expected 1`] = `
<div>
  <p
    class="css-fzcsno"
  >
    Test One
  </p>
  <p
    class="css-fzcsno"
  >
    Test Two
  </p>
  <p
    class="css-fzcsno"
  >
    Test Three
  </p>
</div>
`;

exports[`<Text /> should render with the correct element tag 1`] = `
<div>
  <p
    class="css-fzcsno"
  >
    Paragraph
  </p>
  <i
    class="css-fzcsno"
  >
    Italic
  </i>
  <strong
    class="css-fzcsno"
  >
    Strong
  </strong>
  <span
    class="css-fzcsno"
  >
    Span
  </span>
</div>
`;
